data/mods/ssb/random-teams.ts

```diff
@@ -871,44 +871,53 @@ export class RandomStaffBrosTeams extends RandomTeams {
 	randomStaffBrosTeam(options: {inBattle?: boolean} = {}) {
 		const team: PokemonSet[] = [];
 		const debug: string[] = []; // Set this to a list of SSB sets to override the normal pool for debugging.
-		const pool = debug.length ? debug : Object.keys(ssbSets);
+		const ruleTable = this.dex.formats.getRuleTable(this.format);
+		let pool = debug.length ? debug : Object.keys(ssbSets);
 		const typePool: {[k: string]: number} = {};
 		let depth = 0;
+		let isFiltered = false;
 		while (pool.length && team.length < 6) {
 			if (depth >= 200) throw new Error(`Infinite loop in Super Staff Bros team generation.`);
 			depth++;
 			const name = this.sampleNoReplace(pool);
 			const ssbSet: SSBSet = this.dex.deepClone(ssbSets[name]);
 			if (ssbSet.skip) continue;
-
-			// Enforce typing limits
+			const types = this.dex.species.get(ssbSet.species).types;
 			if (!debug.length) { // Type limits are ignored when debugging
-				const types = this.dex.species.get(ssbSet.species).types;
-				const weaknesses = [];
-				for (const type of this.dex.types.names()) {
-					const typeMod = this.dex.getEffectiveness(type, types);
-					if (typeMod > 0) weaknesses.push(type);
-				}
-				let rejected = false;
-				for (const type of weaknesses) {
-					if (typePool[type] === undefined) typePool[type] = 0;
-					if (typePool[type] >= 3) {
-						// Reject
-						rejected = true;
-						break;
+				if (ruleTable.has('sametypeclause')) {
+					if (!isFiltered) {
+						const monotype = this.sample(types);
+						pool = pool.filter(x => this.dex.species.get(ssbSets[x].species).types.includes(monotype));
```

Comment:
Shouldn't you select randomly from all types, rather than picking a type of the first Pokémon selected? That's how regular Monotype Random Battle does it, and it would simplify the code (and make each type equally likely to be chosen).

Reply:
good point

Reply:
taken care of, and made the code a lot more concise because of it